xyz

```diff
@@ -127,6 +127,24 @@ inc() {
   printf %s "$version"
 }
 
+if type tput &>/dev/null ; then
+  bold=$(tput bold)
+  smul=$(tput smul)
+  rmul=$(tput rmul)
+  reset=$(tput sgr0)
+fi
+
+check_bash_version() {
+  if (( BASH_VERSINFO[0] < 4 )) ; then
+    local xyz
+    xyz="$(IFS=.; echo "${BASH_VERSINFO[*]:0:3}")"
+    abort \
+      "Inadequate Bash version: ${bold}${xyz}${reset} < ${bold}4.0.0${reset}" \
+      "${smul}https://github.com/davidchambers/xyz/issues/51${rmul}"
+  fi
+}
```

Comment:
Implementation notes:

  - `$(IFS=.; echo "${BASH_VERSINFO[*]:0:3}")` joins the first three elements of [`BASH_VERSINFO`][1] with `.`. Note that the subshell is necessary, as this [comment][2] explains.

  - `local xyz="$(...)"` would trigger a ShellCheck warning: [SC2155][3].

  - `abort` is being called with two arguments.


[1]: https://www.gnu.org/software/bash/manual/html_node/Bash-Variables.html#index-BASH_005fVERSINFO
[2]: https://stackoverflow.com/questions/57627024/using-ifs-with-echo#comment101707914_57627024
[3]: https://www.shellcheck.net/wiki/SC2155